### PR TITLE
0064: Hide empty project overview sections

### PR DESCRIPTION
**Pull request by illume**
## Summary

Patch 0064 hides plugin-provided project overview cards when their component
renders no content. Populated plugin sections continue to use the standard
three-column card layout.

## Screenshots

| Before | After |
| --- | --- |
| Before: an empty custom project overview card remains visible | After: the empty custom project overview card is hidden |

The images use the actual overview card component in the project overview grid
at the same 1200 px viewport.

## Source and provenance

- Original change: [Azure/aks-desktop#478](https://github.com/Azure/aks-desktop/pull/478)
- Original commit: [c25799eae65bce8446f8b692a36fc7b5d74549a8](https://github.com/Azure/aks-desktop/commit/c25799eae65bce8446f8b692a36fc7b5d74549a8)
- Retained as patch 0064 by [Azure/aks-desktop#823](https://github.com/Azure/aks-desktop/pull/823)
- Patch-series commit: [387eb27e3f586211610dda61949255e7264a9c95](https://github.com/Azure/aks-desktop/commit/387eb27e3f586211610dda61949255e7264a9c95)

The behavior commit preserves Oleksandr Dubenko as author and committer with
the original 2026-03-19 timestamp, and adds René Dudfield as co-author.

## Changes

- Extract the plugin overview card markup into `ProjectOverviewSection` for
  focused testing.
- Cover populated and empty plugin section output before changing behavior.
- Add an in-cluster Playwright test using the projects example plugin.
- Hide the grid item with CSS when its `CardContent` is empty.

## Improvements over the existing change

- Adds direct unit coverage for both populated and empty section output.
- Scopes empty detection to the wrapper card so nested plugin cards remain visible.
- Reaches 100% branch coverage for the changed overview section renderer.
- Adds end-to-end coverage through a deployed plugin fixture.
- Keeps the tests in commits before the behavior change, while the behavior
  commit updates both unit and e2e assertions to cover the regression.

## Steps to test

1. Register a project overview section whose component returns `null`.
2. Open a project overview page.
3. Confirm populated plugin cards remain visible and no blank card is shown.

Validated locally:

- `npm test -- --run src/components/project/ProjectOverviewSection.test.tsx src/components/project/ProjectDetails.test.tsx` (14 tests)
- Focused Istanbul coverage: 100% statements, branches, functions, and lines
- `npx eslint src/components/project/ProjectOverviewSection.tsx src/components/project/ProjectOverviewSection.test.tsx`
- `npm run tsc`
- `npx playwright test tests/projectOverview.spec.ts --list` (2 tests)

The full Playwright scenario requires the repository's two-cluster in-cluster
fixture and `HEADLAMP_TEST_TOKEN`; local validation confirmed discovery and CI
will execute the deployed scenario.

## Notes for the reviewer

Assisted by copilot.

### CODE REVIEW

File: frontend/src/components/project/ProjectOverviewSection.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  project,
+  projectResources,
+  section,
+}: ProjectOverviewSectionProps) {
+  return (
+    <Grid
+      item
+      xs={12}
+      md={4}
+      sx={{ '&:has(.MuiCardContent-root:empty)': { display: 'none' } }}
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/project/ProjectOverviewSection.tsx]: Addressed in ade26f216. The file is formatted with the repository Prettier configuration, and the exact `npm run frontend:lint` CI entry point now passes.

[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/project/ProjectOverviewSection.tsx]: Addressed in ade26f216. The selector now targets only the wrapper chain (`> .MuiCard-root > .MuiCardContent-root:empty`), and a regression test verifies that visible plugin content remains shown when the plugin renders an empty nested MUI card. Frontend lint, TypeScript, six focused tests, and 100% renderer branch coverage pass.

**Comment by illume** [comment]
rebased on main, fixed git conflicts

**Review by unlikelyzero — CHANGES_REQUESTED** [review]
Nice cleanup pulling this into its own component, and the nested-empty-card fix from the earlier review comment looks solid — I reverted the selector locally and confirmed the new unit test actually catches that regression.

A few things worth addressing before merge:

1. `ProjectOverviewSection.tsx:45` — The `:empty` CSS selector only hides a section when `section.component` returns literal `null`. A plugin returning a non-null empty wrapper (e.g. `<Box>{items.map(...)}</Box>` with an empty array) still inserts a DOM element into `CardContent`, so `:empty` won't match and the user sees a blank card — the exact problem this PR is meant to fix. The added test/example fixture only covers `component: () => null`, so this gap isn't caught. Worth documenting the `return null` contract explicitly in the `registerProjectOverviewSection` JSDoc, and adding a regression test for a non-null empty wrapper.

2. `ProjectOverviewSection.tsx:50` — `section.component` renders without an `ErrorBoundary`, unlike every other plugin extension point in the app (see `TopBar.tsx`, `AppLogo.tsx`, `Chooser.tsx`). `display: none` only applies after the component has rendered, so a section that throws still takes down the whole page on its way to being hidden. Wrapping it in `ErrorBoundary` would also compose nicely with this PR's own hiding logic — a crashing section degrades to invisible instead of breaking the page.

3. `e2e-tests/tests/projectOverview.spec.ts:118` — The assertion counts every empty-looking card on the page (`toHaveCount(1)`) rather than targeting the `empty-section` fixture specifically. If any of the built-in overview cards ever renders empty for an unrelated reason this fails for the wrong reason, and it wouldn't catch the fixture failing to register at all. A `data-testid` on the fixture's grid item would make this more targeted.

4. Smaller ones, non-blocking:
   - The visibility selector hardcodes `.MuiCard-root`/`.MuiCardContent-root` in both the component and the e2e test — using MUI's exported `cardClasses`/`cardContentClasses` constants would avoid both breaking silently on a MUI version bump.
   - Hiding a section only after its async content resolves introduces a new whole-card layout shift that didn't exist before (previously the card was always present, so only its inner content shifted).
   - The new `ProjectOverviewSection` component name collides with the existing exported `ProjectOverviewSection` type in `redux/projectsSlice.ts`, forcing an alias in a couple of places — might be worth a more specific name like `ProjectOverviewSectionCard`.
   - `projectOverview.spec.ts` is missing its trailing newline (fails `prettier --check`, though nothing in e2e-tests currently lints for it).
   - The new `empty-section` registration in the example plugin has no comment explaining it's an e2e fixture.

**Review by ashu8912 — APPROVED** [review]
